torch_geometric/explain/algorithm/gnn_explainer.py

```diff
@@ -76,6 +77,34 @@ def __init__(self, epochs: int = 100, lr: float = 0.01, **kwargs):
 
         self.node_mask = self.edge_mask = None
 
+    def supports(
+        self,
+        explainer_config: ExplainerConfig,
+        model_config: ModelConfig,
```

Comment:
We can address this in another PR. 
What if `explainer_config` and `model_config` were inputs to `ExplainerAlgorithm.__init__()`, and we would call `self.support` while initializing the class, feels like thats more cleaner.